File: arxiv_test.ts
//
// Test our ArXiv methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizeArXivID, validateArXivID } from "./arxiv.ts";
import { verifyArXivID } from "./arxiv_record.ts";

const varified_ids: string[] = [
  "arXiv:2412.03631",
  "arXiv:2412.03649",
  "arXiv:2412.04386",
];

for (const id of varified_ids) {
  // Normalize
  const normalized = normalizeArXivID(id);
  assertNotEquals(normalized, undefined);
  // Validate
  assertEquals(validateArXivID(id), true);
  // Verify is via arxiv.org
  const isOK: boolean = await verifyArXivID(id);
  assertEquals(isOK, true);
}

File: arxiv_test.go
package metadatatools

import "testing"

func TestArXivID(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"arXiv:2412.03631",
		"arXiv:2412.03649",
		"arXiv:2412.04386",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizeArXivID(id)
		if normalized == "" {
			t.Errorf("NormalizeArXivID(%q) returned an empty string", id)
		}
		t.Logf("Normalized arXiv ID: %s", normalized)

		if !ValidateArXivID(id) {
			t.Errorf("ValidateArXivID(%q) = false, expected true", id)
		}

		if !VerifyArXivID(id) {
			t.Errorf("VerifyArXivID(%q) = false, expected true", id)
		}
	}
}
